src/title.js

```diff
@@ -0,0 +1,437 @@
+		// </title>
+		//
+		// See: https://github.com/ckeditor/ckeditor5/issues/2005.
+		model.schema.register( 'title', { inheritAllFrom: '$block' } );
+		model.schema.register( 'title-content', {
+			inheritAllFrom: '$block',
+			allowIn: 'title'
+		} );
+
+		model.schema.addChildCheck( ( context, childDefinition ) => {
+			// Allow `title` element only directly inside a root.
+			if ( childDefinition.name === 'title' && !context.endsWith( '$root' ) ) {
+				return false;
+			}
+
+			// Allow only `title-content` inside `title`.
+			if (
+				context.endsWith( 'title' ) && childDefinition.name !== 'title-content' ||
+				childDefinition.name === 'title-content' && !context.endsWith( 'title' )
+			) {
+				return false;
+			}
+		} );
+
+		// Disallow all attributes in `title-content`.
+		model.schema.addAttributeCheck( context => {
+			if ( context.endsWith( 'title-content $text' ) ) {
+				return false;
+			}
+		} );
+
+		// Because of `title` is represented by two elements in the model
+		// but only one in the view it's needed to adjust Mapper.
+		editor.editing.mapper.on( 'modelToViewPosition', mapModelPositionToView( editor.editing.view ) );
+		editor.data.mapper.on( 'modelToViewPosition', mapModelPositionToView( editor.editing.view ) );
+
+		// `title-content` <-> `h1` conversion.
+		editor.conversion.elementToElement( { model: 'title-content', view: 'h1' } );
+
+		// Take care about proper `title` element structure.
+		model.document.registerPostFixer( writer => this._fixTitleContent( writer ) );
+
+		// Create and take care about proper position of a `title` element.
+		model.document.registerPostFixer( writer => this._fixTitleElement( writer ) );
+
+		// Prevent from adding extra paragraph after paste or enter.
+		model.document.registerPostFixer( writer => this._preventExtraParagraphing( writer ) );
+
+		// Attach `Title` and `Body` placeholders to the empty title and/or content.
+		this._attachPlaceholders();
+
+		// Attach Tab handling.
+		this._attachTabPressHandling();
+	}
+
+	/**
+	 * Sets the title of the document. This methods does not change any content outside the title element.
+	 *
+	 * @param {String} data Data to be set as a document title.
+	 */
+	setTitle( data ) {
+		const editor = this.editor;
+		const titleContent = editor.model.document.getRoot().getChild( 0 ).getChild( 0 );
+
+		editor.model.insertContent( editor.data.parse( data, 'title-content' ), titleContent, 'in' );
+	}
+
+	/**
+	 * Returns the title of the document. Note, that because this plugin does not allow any formatting inside
+	 * the title element, the output of this method will be a plain text, with no HTML tags. However, it
+	 * may contain some markers, like comments or suggestions. In such case, a special tag for the
+	 * marker will be included in the title text.
+	 *
+	 * @returns {String} Title of the document.
+	 */
+	getTitle() {
+		const title = this.editor.model.document.getRoot().getChild( 0 ).getChild( 0 );
```

Comment:
I know that we have a post-fixer but I would feel safer with using `getTitleElement()`, WDYT?

Reply:
Or OTOH - maybe add a comment about that post-fixer takes care of this.

Reply:
hm..... I was thinking about using `getTitleElement()` but finally decided to do it as it is. `getTitleElement()` it's an additional call of `Array.from` and `Array.find`. As you wrote, post-fixer takes care of this structure.

Reply:
I changed this.